src/nomad_parser_magres/parsers/parser.py

```diff
@@ -0,0 +1,740 @@
+            magres_data=magres_data, cell=cell, logger=logger
+        )
+        if len(isc) > 0:
+            outputs.spin_spin_couplings = isc
+
+        # Parse `MagneticSusceptibility`
+        mag_sus = self.parse_magnetic_susceptibilities(
+            magres_data=magres_data, logger=logger
+        )
+        if len(mag_sus) > 0:
+            outputs.magnetic_susceptibilities = mag_sus
+
+        return outputs
+
+    def parse_nmr_magres_file_format(
+        self, nmr_first_principles_archive: 'EntryArchive'
+    ):
+        """
+        Automatically parses the NMR Magres workflow. Here, `self.archive` is the
+        NMR magres archive in which we will link the original NMR first principles (CASTEP
+        or QuantumESPRESSO) entry.
+
+        Args:
+            nmr_first_principles_archive (EntryArchive): the NMR (first principles) CASTEP or QuantumESPRESSO archive.
+        """
+        workflow = NMRMagRes(method=NMRMagResMethod(), results=NMRMagResResults())
+        workflow.name = 'NMR Magres'
+
+        # ! Fix this once CASTEP and QuantumESPRESSO use the new `nomad-simulations` schema under 'data'
+        # Method
+        # method_nmr = extract_section(nmr_first_principles_archive, ['run', 'method'])
+        # workflow.method.nmr_method_ref = method_nmr
+
+        # Inputs and Outputs
+        # ! Fix this to extract `input_structure` from `nmr_first_principles_archive` once
+        # ! CASTEP and QuantumESPRESSO use the new `nomad-simulations` schema under 'data'
+        input_structure = extract_section(self.archive, ['data', 'model_system'])
+        nmr_magres_calculation = extract_section(self.archive, ['data', 'outputs'])
+        if input_structure:
+            workflow.m_add_sub_section(
+                NMRMagRes.inputs, Link(name='Input structure', section=input_structure)
+            )
+        if nmr_magres_calculation:
+            workflow.m_add_sub_section(
+                NMRMagRes.outputs,
+                Link(name='Output NMR calculation', section=nmr_magres_calculation),
+            )
+
+        # NMR (first principles) task
+        # ! Fix this once CASTEP and QuantumESPRESSO use the new `nomad-simulations` schema under 'data'
+        program_name = nmr_first_principles_archive.run[-1].program.name
+        if nmr_first_principles_archive.workflow2:
+            task = TaskReference(task=nmr_first_principles_archive.workflow2)
+            task.name = f'NMR FirstPrinciples {program_name}'
+            if input_structure:
+                task.inputs = [Link(name='Input structure', section=input_structure)]
+            if nmr_magres_calculation:
+                task.outputs = [
+                    Link(
+                        name='Output NMR calculation',
+                        section=nmr_magres_calculation,
+                    )
+                ]
+            workflow.m_add_sub_section(NMRMagRes.tasks, task)
+
+        self.archive.workflow2 = workflow
```

Comment:
At the moment, the workflow separates the .castep (NMR first principles calculation) file and .magres file resulting from the calculation as two individual entries, with both having display names "<formula> CASTEP NMR SinglePoint Simulation". Ideally, since the magres file is a direct output of the CASTEP calculation, it might be beneficial to include it directly to the outputs of the CASTEP first principles calculation in the workflow.

Will the "fix" mentioned in the code comments correspond to fixing how the workflow is processed currently? Or is it related only to the adoption of the new 'nomad-simulations' schema?